### PR TITLE
OpenCL: rev support for trigonometric functions

### PR DESCRIPTION
**Pull request by rok-cesnovar**
## Summary

This PR adds /rev OpenCL support for acos, acosh, asin, asinh, atan, atanh, cos, cosh, sin, sinh, tan, tanh

## Tests

/

## Side Effects

/

## Release notes

OpenCL: added /rev support for acos, acosh, asin, asinh, atan, atanh, cos, cosh, tan, tanh

## Checklist

- [x] Math issue #1854 

- [x] Copyright holder: Rok Češnovar, Uni. of Ljubljana

### CODE REVIEW

File: stan/math/opencl/rev/tanh.hpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * @param A argument
+ * @return Elementwise `tanh()` of the input, in radians.
+ */
+inline var_value<matrix_cl<double>> tanh(
+    const var_value<matrix_cl<double>>& A) {
+  var_value<matrix_cl<double>> res = tanh(A.val());
+
+  reverse_pass_callback([A, res]() mutable {
+    const matrix_cl<double>& cosh_A_val = cosh(A.val());
```
Comment (t4c1) [inline stan/math/opencl/rev/tanh.hpp:24]:
```suggestion
    auto cosh_A_val = cosh(A.val());
```
No need to eval it in a separate kernel. Kernel generator will deduplicate iit, so cosh will only be calculated once.

File: test/unit/math/opencl/rev/acos_test.cpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#ifdef STAN_OPENCL
+#include <stan/math/opencl/rev.hpp>
+#include <stan/math.hpp>
+#include <gtest/gtest.h>
+#include <test/unit/math/opencl/util.hpp>
+
+auto acos_functor = [](const auto& a) { return stan::math::acos(a).eval(); };
```
Comment (t4c1) [inline test/unit/math/opencl/rev/acos_test.cpp:7]: No need for eval.
```suggestion
auto acos_functor = [](const auto& a) { return stan::math::acos(a); };
```

Comment (t4c1) [reply test/unit/math/opencl/rev/acos_test.cpp:7]: Same for other tests.

**Review by t4c1 — CHANGES_REQUESTED** [review]

File: test/unit/math/opencl/rev/acos_test.cpp (hunk 2 above)
Comment (rok-cesnovar) [reply test/unit/math/opencl/rev/acos_test.cpp]: Done.

**Review by t4c1 — APPROVED** [review]

**Review by t4c1 — APPROVED** [review]

**Pull request merged**